src/main/network/http.ts

```diff
@@ -453,7 +454,53 @@ export const httpGetWithAuth =
 
         };
 
-export const httpGet = httpGetWithAuth(true);
+const httpGetFactory =
+    (): typeof httpFetchFormattedResponse =>
+        async (...arg) => {
+            const [_url, _options, _callback, ..._arg] = arg;
+
+            let url: URL | undefined;
+            try {
+                url = new URL(_url);
+            } catch (e) {
+                // wrong URL : fallback to httpGetWithAuth
+                debug("wrong URL : fallback to httpGetWithAuth");
+                debug(e);
+            }
+
+            if (url?.protocol === "file:") {
+
+                let isFailure = false;
+                let stream: NodeJS.ReadableStream | undefined;
+                try {
+                    stream = createReadStream(url.pathname);
```

Comment:
Enormous security risk (full disk access).

Reply:
So, a maliciously-crafted OPDS feed, LCP license, etc. could trigger arbitrary filesystem location access. This wouldn't pass a security audit, even if in practice the recipient of the HTTP response would be limited to some internal Thorium API, because an arbitrary filesystem path could be an operating system hook (not necessarily a downloadable binary blob). In other words, a mounted filesystem should be regarded as a sensitive API surface, and Thorium should implement provisions to secure access (e.g. ask user permissions, limit to known folders, etc.).

Reply:
https://code.visualstudio.com/docs/editor/workspace-trust

Reply:
should we used the same mechanism ?